src/stores/room-list/algorithms/list_ordering/Algorithm.ts

```diff
@@ -0,0 +1,177 @@
+/*
```

Comment:
Why is the directory `list_ordering` instead of `list-ordering`...? I think we use hyphens most often as separators...?

Reply:
probably only because I hit the keyboard. There's no justifiable reason for it.